src/app/modules/main/communities/tokens/module.nim

```diff
@@ -80,23 +80,33 @@ proc authenticate(self: Module) =
   else:
     self.controller.authenticateUser()
 
-method airdropCollectibles*(self: Module, communityId: string, collectiblesJsonString: string, walletsJsonString: string) =
-  let collectiblesJson = collectiblesJsonString.parseJson
-  self.tempTokenAndAmountList = @[]
-  for collectible in collectiblesJson:
-    let symbol = collectible["key"].getStr
-    let amount = collectible["amount"].getInt
-    let tokenDto = self.controller.getCommunityTokenBySymbol(communityId, symbol)
-    if tokenDto.tokenType == TokenType.Unknown:
-      error "Can't find token for community", communityId=communityId, symbol=symbol
-      return
-    self.tempTokenAndAmountList.add(CommunityTokenAndAmount(communityToken: tokenDto, amount: amount))
+proc getTokenAndAmountList(self: Module, communityId: string, collectiblesJsonString: string): seq[CommunityTokenAndAmount] =
+  try:
+    let collectiblesJson = collectiblesJsonString.parseJson
```

Comment:
Does it make sense to make it as generic naming since I guess in case of assets it would be the same data.. `contractUniqueKey` and `amount` too?